### PR TITLE
Docker Toolbox on Windows – notes

### PR DESCRIPTION
**Pull request by borekb**
Part of #1389.

I'm verifying how `tests:full` run on Docker Toolbox for Windows, this PR documents the process.

See:

- My setup from clean Windows machine: https://github.com/versionpress/versionpress/pull/1403#issuecomment-473608973
- Updated docs: https://github.com/versionpress/versionpress/pull/1403/commits/a13ba8f96a0aa98cf5abe0047e45235778ce6367

### CODE REVIEW

**Comment by borekb** [comment]
The tests are at least running!

image

I'm testing on an old Windows laptop (~2014 HP ProBook 650) that I wiped and updated to the latest Windows 10 version. These were the steps I then followed:

Install VSCode.

Install Git for Windows. (I generally prefer [MSYS2](https://medium.com/@borekb/zsh-via-msys2-on-windows-3964a943b1ce) but wanted to stay closer to what most people would run.)

- Select "Use Git and optional Unix tools..." option.
- Select "Check out as-is, commit as-is" option.

Do a rudimentary Git setup (copy `.gitconfig` from GitHub etc.).

Install ConEmu, a sane terminal.

Install Docker Toolbox.

Install Node.js.

The I remembered that [Scoop](https://scoop.sh/) exists so I installed that.

Install PHP (v7.3 at the time of writing):

```
scoop install php-nts
```

Enable PHP extension by following this: https://github.com/lukesampson/scoop/wiki/Custom-PHP-configuration.

Install Composer:

```
scoop install composer
```

Then I was able to start the _Docker Quickstart Terminal_, configure port forwarding and folder sharing (see updated dev-setup.md in https://github.com/versionpress/versionpress/pull/1403/commits/a13ba8f96a0aa98cf5abe0047e45235778ce6367) and run tests. Unit tests passed:

```
npm run tests:unit
```

**Comment by borekb** [comment]
Great, I'm getting consistent results with Docker for Mac, https://github.com/versionpress/versionpress/issues/1400#issuecomment-471987716:

image

It took about half an hour, maybe a bit more, which is again quite similar to what I'm seeing on Mac, currently (I can no longer reproduce the 10-minute runs, see https://github.com/versionpress/versionpress/issues/1400#issuecomment-473606507).

**Comment by borekb** [comment]
Waiting is implemented in #1401 if that was part of the problem...

@pavelevap, if you have more tips or notes to share, feel free to add them, otherwise, I'm going to merge this later today.

**Comment by borekb** [comment]
Also worth noting is that Travis now supports Windows so it would be a good idea to run tests on Windows automatically (though it most likely won't be via Toolbox). See https://github.com/versionpress/versionpress/issues/1259#issuecomment-473793704.

File: docs/content/en/developer/dev-setup.md [hunk 1]
```diff
@@ -178,15 +178,16 @@ As noted in [Getting started](#getting-started), we only support Git Bash on Win
...
+If you can, use [Docker Desktop](https://www.docker.com/products/docker-desktop), not [Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_windows/). The experience will be generally smoother.
 
 If you need to use Docker Toolbox:
 
-- Enable port forwarding in VirtualBox (especially for ports 80, 3306, 8080 and 8099), see [details](https://stackoverflow.com/questions/42866013/docker-toolbox-localhost-not-working/45822356#45822356).
-- Docker Toolbox is slower, but you can try to adjust system performance in VirtualBox settings. However, you can run into timeout issues in Workflow tests sometimes and exceed default value of 5 seconds in `wp_remote_get()` in End2End tests.
-- Run Git Bash and Docker Quickstart Terminal as an Administrator to avoid potential problems (for example permissions and symlinks).
+1. [Enable port forwarding in VirtualBox](https://stackoverflow.com/questions/42866013/docker-toolbox-localhost-not-working/45822356#45822356), especially for ports 80 and 3306.
```
Comment (pavelevap) [inline docs/content/en/developer/dev-setup.md]: Also port 8099 is needed for access to Adminer.

Comment (borekb) [reply docs/content/en/developer/dev-setup.md]: Resolved in 65b247e.

**Comment by borekb** [comment]
I've updated the section on Docker Toolbox performance after my recent experiments in https://github.com/versionpress/versionpress/pull/1401#issuecomment-476004709.

**Comment by borekb** [comment]
On the laptop (2014 HP ProBook 650), I upgraded from Docker Toolbox to Docker Desktop (had to completely reinstall the machine as Hyper-V was failing to activate after VirtualBox had been previously installed 🙄).

The tests ran for a much more reasonable 16 minutes - it was 37 before and the best times we're getting on powerful MacBooks or on TravisCI are about 11 minutes. So not too bad.

image

**Pull request merged**